core/src/test/java/org/jdbi/v3/BeanMapperTest.java

```diff
@@ -61,6 +61,23 @@ public void shouldSetValueOnPublicSetter() throws Exception {
     }
 
     @Test
+    public void shouldHandleColumNameWithUnderscores() throws Exception {
+        expect(resultSetMetaData.getColumnCount()).andReturn(1).anyTimes();
+        expect(resultSetMetaData.getColumnLabel(1)).andReturn("LONG_FIELD");
```

Comment:
I added a test to check for both underscore syntax and case insensitivity in 3012ebb.

The two other proposed input (longField and LongField) are actually already covered in other tests of the same class.